src/telliot_ampl_feeds/sources/usd_vwap.py

```diff
@@ -0,0 +1,206 @@
+import asyncio
+import statistics
+from abc import ABC
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any
+from typing import List
+from typing import Mapping
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+from typing import TypeVar
+
+import requests
+from telliot.datasource import DataSource
+from telliot.types.datapoint import datetime_now_utc
+from telliot.types.datapoint import OptionalDataPoint
+from telliot.utils.response import ResponseStatus
+
+from telliot_ampl_feeds.config import AMPLConfig
+
+
+T = TypeVar("T")
+
+
+async def get_float_from_api(
+    url: str,
+    params: Sequence[Any],
+    headers: Optional[Mapping[str, str]] = None,
+) -> OptionalDataPoint[float]:
+    """Helper function for retrieving datapoint values."""
+
+    with requests.Session() as s:
+        try:
+            r = None
+            if headers:
+                r = s.get(url, headers=headers)
+            else:
+                r = s.get(url)
+            data = r.json()
+
+            for param in params:
+                data = data[param]
+
+            timestamp = datetime_now_utc()
+            datapoint = (data, timestamp)
+
+            return datapoint
+
+        except requests.exceptions.ConnectTimeout:
+            return (None, None)
+
+        except Exception:
+            return (None, None)
+
+
+@dataclass
+class AnyBlockSource(DataSource):
+    """Data source for retrieving AMPL/USD/VWAP from AnyBlock api."""
+
+    api_key: str = ""
+
+    async def fetch_new_datapoint(
+        self,
+    ) -> OptionalDataPoint[float]:
+        """Update current value with time-stamped value."""
+
+        url = (
+            "https://api.anyblock.tools/market/AMPL_USD_via_ALL/daily-volume"
+            + "?roundDay=false&debug=false&access_token="
+            + self.api_key
+        )
+        params = ["overallVWAP"]
+
+        datapoint = await get_float_from_api(url, params)
+
+        self.store_datapoint(datapoint)
+
+        return datapoint
+
+
+@dataclass
+class BraveNewCoinSource(DataSource):
```

Comment:
A little curious why mypy doesn't want to see:
`class BraveNewCoinSource(DataSource[float]):`

Reply:
It does now, I wasn't running it with `--strict`